api/app/controllers/spree/api/v2/storefront/menus_controller.rb

```diff
@@ -13,7 +13,7 @@ def serialize_collection(collection)
           end
 
           def resource
-            scope.find_by(unique_code: params[:unique_code])
+            @resource ||= scope.find_by(location: params[:location])
```

Comment:
@MatthewKennedy oh last thing, I think we also need to include locale in the scope, eg.

```ruby
def scope
  super.by_store(current_store).by_locale(I18n.locale)
end
```